src/marketplace/ListingsView.js

```diff
@@ -32,117 +38,171 @@ class ListingsView extends React.Component<Props, State> {
 		super(props);
 		this.state={
 			userId: '',
+			chassisForSale: [],
+			weaponsForSale: [],
+			scannersForSale: [],
+			scannerAddonsForSale: [],
+			jammersForSale: [],
+			treadsForSale: [],
 			itemsForSale: [],
-			tanksForSale: null,
+			tanksForSale: [],
+			tanks: [],
+			cards: [<div key={0}><h5>Loading Sales...</h5></div>]
 		}
 	}
 
-	// Once mounted, get the user's ID.
-	componentDidMount() {
-		this.getUserId();
+	// Once mounted, get the user's ID and set the sales.
+	componentDidMount(): void {
+		getUserAPICall(user => {
+			this.setState({userId: user.userId});
+			this.getSales();
+		});
 	}
 
-	// When sellerType is updated we need to get the new sells
+	// When updating, update the cards according to the new seller type.
 	componentDidUpdate(prevProps: Props, prevState: State): void {
-		if(prevProps !== this.props) {
-			this.directSaleToProperFunction();
-		}
-	}
-
-	// This function directs the view to the proper function
-	// If we are in the tank view it directs it to getMarketSalesForTanks()
-	// Else it directs it to getMarketSalesForComponents()
-	directSaleToProperFunction(): void {
-		if(this.props.sellerType === 'tanks') {
-			this.getMarketSalesForTanks();
-		}
-		else {
-			this.getMarketSalesForComponents();
+		if (prevProps !== this.props) {
+			switch (this.props.sellerType) {
+				case 'chassis':
```

Comment:
Creating these cards should be done in the render method and they shouldn't be stored inside the component's state

Reply:
Once I am back, I will change it to map the cards out inside the render method instead. It might take me a bit to change stuff around to make it work, but it can't be too hard.